numba/cuda/tests/cudapy/test_record_dtype.py

```diff
@@ -403,6 +419,20 @@ def test_record_read_2d_array(self):
         res = cfunc(nbval[0])
         np.testing.assert_equal(res, nbval[0].j[1, 0])
 
+    def test_setitem(self):
+        nbarr1 = np.recarray(2, dtype=recordwith2darray)
+        nbarr1[0] = np.array([(1, ((1, 2), (4, 5), (2, 3)))],
+                             dtype=recordwith2darray)[0]
+        nbarr2 = np.recarray(2, dtype=recordwith2darray)
+        nbarr2[0] = np.array([(10, ((10, 20), (40, 50), (20, 30)))],
+                             dtype=recordwith2darray)[0]
+        pyfunc = record_setitem_array
+        args = nbarr1[0], nbarr2[0]
+        arr_expected = pyfunc(*args)
+        cfunc = cuda.jit(pyfunc)
+        arr_res = cfunc[1, 1](*args)
+        np.testing.assert_equal(arr_res, arr_expected)
```

Comment:
This suffers from the same testing `None` == `None` issue as noted here: https://github.com/numba/numba/pull/7712/files#r789550428?

Reply:
Good point - I've just added a change modelled on your fix that resolves this.